app/queries/servers/channel.ts

```diff
@@ -50,25 +51,49 @@ export function prepareMissingChannelsForAllTeams(operator: ServerDataOperator,
     }
 }
 
+type MembershipWithId = ChannelMembership & {id: string};
+
+type MembershipReduce = {
+    memberships: MembershipWithId[];
+    membershipsMap: Record<string, MembershipWithId>;
+}
+
 export const prepareMyChannelsForTeam = async (operator: ServerDataOperator, teamId: string, channels: Channel[], channelMembers: ChannelMembership[]) => {
-    const allChannelsForTeam = await queryAllChannelsForTeam(operator.database, teamId).fetch();
+    const {database} = operator;
+    const allChannelsForTeam = (await queryAllChannelsForTeam(database, teamId).fetch()).reduce((map, channel) => {
```

Comment:
Same about using the generic of reduce, instead of `as`. Since I see it in several places, I won't flag it any more.